os/mm/umm_heap/umm_calloc.c

```diff
@@ -106,12 +106,19 @@ void *calloc_at(int heap_index, size_t n, size_t elem_size)
 
 FAR void *calloc(size_t n, size_t elem_size)
 {
-	int heap_idx;
+	int heap_idx = 0;
+	int prio_idx = 0;
 	void *ret;
+
+#ifdef CONFIG_RAM_MALLOC_PRIOR_INDEX
+	heap_idx = CONFIG_RAM_MALLOC_PRIOR_INDEX;
+	prio_idx = (CONFIG_RAM_MALLOC_PRIOR_INDEX > 0) ? CONFIG_RAM_MALLOC_PRIOR_INDEX : 0;
```

Comment:
I think `prio_idx = CONFIG_RAM_MALLOC_PRIOR_INDEX;` is simple and enough.
User should guarantee it's a nonnegative value.

Reply:
refined the logic, and use macro for new code, to avoid unnecessary text size increase